src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1558,10 +1558,9 @@ private DateTime GetDateHeaderHelper(string headerName)
 
         private void SetDateHeaderHelper(string headerName, DateTime dateTime)
         {
-            if (dateTime == DateTime.MinValue)
-                SetSpecialHeaders(headerName, null); // remove header
-            else
-                SetSpecialHeaders(headerName, HttpDateParser.DateToString(dateTime.ToUniversalTime()));
+            SetSpecialHeaders(headerName, dateTime == DateTime.MinValue ?
+                null : // remove header
+                dateTime.ToString("r"));
```

Comment:
`DateTime.ToUniversalTime().ToString("r")` and `DateTime.ToString("r")` produce different results. I think the call still needs to remain here since we're no longer bouncing through the `DateTimeOffset` indirection, which would normally account for this.

Reply:
You're right.  Though it really seems like a (documented) bug that a local DateTime.ToString("r") outputs a time that says "GMT" but that isn't.